scroll-animations-1/Overview.bs

```diff
@@ -487,21 +492,37 @@ The procedure to <dfn>resolve a container-based offset</dfn> given
     first matching condition from the following:
     <div class="switch">
 
-    :   <var>offset</var> is <code>auto</code>
+    :   If <var>offset</var> is <code>auto</code>:
     ::  Either the beginning or the ending of {{source}}'s scroll range in
         {{orientation}} depending on whether the offset is {{start}} or {{end}}.
 
-    :   <var>offset</var> is a <<length-percentage>>
+    :   If <var>offset</var> is a {{CSSNumberish}} and matches <<number>>:
+    ::  The distance indicated by the value in <a value>pixel unit</a> along
```

Comment:
I was under the impression that treating plain numbers as `px` was a pattern we didn't want? Maybe get someone else's opinion on this. (@tabatkins ?)

Reply:
I personally felt allowing numbers to be interpreted as `px` would make the API easier to use in Javascript. Specially as it matches all legacy scroll related APIs where numbers are treated as pixels


But I can also see at least two possible concerns:
 - Automatic type conversion can be a footgun but perhaps this is limited enough to be safe
 - Having every downstream spec convert `<numbers>` differently would be bad. So perhaps have a common way to do this i.e., `<number>` -> `<length>` with pixel for all JS API's that accept length but want to accept number too.

BTW, does TypedOM have a special handling for `0` length for which in CSS the unit is optional?

In any case, this is a trade-off and @tabatkins expert's opinion is appreciated. :)

Reply:
We really don't want to interpret bare numbers as px. The definition of CSSNumberish interprets numeric values as a CSS "number" not a length. `offset` should simply be a CSSNumericValue or a CSSStyleValue.

Reply:
Done. I removed CSSNumberish and just use CSSNumericValue which I think is the right type.

Also added a procedure to validate the incoming values when they are being set.